src/concurrency/genmc/mod.rs

```diff
@@ -622,15 +622,15 @@ impl GenmcCtx {
             !self.get_alloc_data_races(),
             "memory deallocation with data race checking disabled."
         );
-        if self
+        let free_result =  self
             .handle
             .borrow_mut()
             .pin_mut()
-            .handle_free(self.active_thread_genmc_tid(machine), address.bytes())
-        {
+            .handle_free(self.active_thread_genmc_tid(machine), address.bytes());
+        if let Some(error) = free_result.as_ref() {
             // FIXME(genmc): improve error handling.
             // An error was detected, so we get the error string from GenMC.
```

Comment:
"so we get the error string from GenMC" isn't the case any more, so the last line can be removed.

(Github does not support code suggestions for commit-by-commit review. :(  )